infra/enrolment-records/repository/src/main/java/com/simprints/infra/enrolment/records/repository/local/migration/InsertRecordsDuringMigrationUseCase.kt

```diff
@@ -0,0 +1,28 @@
+package com.simprints.infra.enrolment.records.repository.local.migration
+
+import com.simprints.infra.config.store.models.Project
+import com.simprints.infra.enrolment.records.repository.domain.models.SubjectAction
+import com.simprints.infra.enrolment.records.repository.local.RoomEnrolmentRecordLocalDataSource
+import com.simprints.infra.logging.LoggingConstants.CrashReportTag.REALM_DB_MIGRATION
+import com.simprints.infra.logging.Simber
+import javax.inject.Inject
+
+// Should be completely removed when all users are migrated to Room
+internal class InsertRecordsDuringMigrationUseCase @Inject constructor(
+    private val realmToRoomMigrationFlagsStore: RealmToRoomMigrationFlagsStore,
+    private val roomEnrolmentRecordLocalDataSource: RoomEnrolmentRecordLocalDataSource,
+) {
+    suspend operator fun invoke(
+        subjectAction: SubjectAction.Creation,
+        project: Project,
+    ) {
+        // if the realm to room migration is in progress, we need to insert the records in the new db
```

Comment:
What if the migration fails?

Reply:
Ah, I get it now - the new records will go to Realm and we also want to insert them in Room, too.
Then I guess the comment can be made clearer:
"// if the realm to room migration is in progress, we need to insert the records in the new db**, too**"